lumapps/api/client.py

```diff
@@ -92,13 +84,10 @@ def f():
             vals = self.cache.get(k)
             if vals:
                 return vals
-            try:
-                token, expiry = self._get_token_and_expiry(email)
-            except HTTPStatusError as err:
-                if err.response.status_code == 403:
-                    raise GetTokenError(get_http_err_content(err))
-                raise
-            self.cache.set(k, (token, expiry), int(expiry - time() - 180))
+            token, expiry = fetch_access_token(
+                self.client, self.base_url, self._auth_info, self.customer_id, email
+            )
+            self.cache.set(k, (token, expiry), expiry)
```

Comment:
shouldn't we expire the token a little early to make it easier to avoid 401 ?
FMI : what is expiry ? is it an amount of time ? or the timestamp of the expiration time ?

Reply:
We could, but in case of `401`, the request will automatically fetch a new token and re-try the request, so it doesn't matter.
`expiry` is the `expires_in` is the number of seconds before the token expires as in https://datatracker.ietf.org/doc/html/rfc6749#section-4.2.2

Reply:
I could substract 30s to the `expiry` value to the cache expiration I suppose :shrug: . Will update

Reply:
great ! :)